backend/de.metas.banking/de.metas.banking.camt53/src/main/java/de/metas/banking/camt53/wrapper/v02/BatchReportEntry2Wrapper.java

```diff
@@ -116,7 +118,10 @@ public Optional<Money> getInterestAmount()
 	@NonNull
 	public Optional<BigDecimal> getCurrencyRate()
 	{
-		return getEntryTransaction()
+		return Stream.of(getEntryTransaction())
+				.filter(list -> !list.isEmpty())
+				.map(list -> list.get(0))
```

Comment:
i see this construction all over:
```java
.filter(list -> !list.isEmpty())
.map(list -> list.get(0))
```

why not introducing a new method called getWhatEverIsFirstLineOrNull(list)

so there you could write it as
```java
.map(this::getWhatEverIsFirstLineOrNull)
.filter(Objects::nonNull)
```